lib/ui/common/roundedButton.dart

```diff
@@ -2,125 +2,156 @@ import 'package:flutter/material.dart';
 
 class RoundedButtonViewModel {
   String title;
-  IconData icon;
+  String icon;
   Function onTap;
-  BuildContext context;
 
-  RoundedButtonViewModel(this.context, {this.title, this.icon, this.onTap});
+  RoundedButtonViewModel({this.title, this.icon, this.onTap});
 }
 
-RoundedButtonViewModel buildButtonViewModel(BuildContext context, {String title, IconData icon, Function onTap}) {
-  return RoundedButtonViewModel(context, title: title, icon : icon, onTap: () => onTap(context));
-}
-
-abstract class RoundedButtonStyle {
-
+class RoundedButtonStyle {
   final Color iconButtonColor;
   final Color backgroundColor;
-
   final BoxShape buttonShape;
   final BorderRadius borderRadius;
-
   final EdgeInsets edgePadding;
   final TextStyle buttonTextStyle;
-
-  final double size;
+  final double height;
+  final double width;
   final double iconEdgePadding;
   final double buttonIconSize;
 
-  RoundedButtonStyle(this.size,
-      this.iconButtonColor, this.backgroundColor, this.iconEdgePadding,
-      {this.buttonShape, this.borderRadius, this.edgePadding,
-        this.buttonTextStyle, this.buttonIconSize});
-}
-
-class _DefaultStyle implements RoundedButtonStyle {
-
-  final Color iconButtonColor =  const Color(0xFFFFFFFF);
-  final Color backgroundColor =  const Color(0xff24d900);
-
-  final BoxShape buttonShape = BoxShape.rectangle;
-  final BorderRadius borderRadius = const BorderRadius.all(Radius.circular(20));
-
-  final EdgeInsets edgePadding = const EdgeInsets.only(left: 32, top: 31, right: 32, bottom: 32);
-  final TextStyle buttonTextStyle = const TextStyle(fontSize: 17, fontWeight: FontWeight.bold, color: Color(0xffffffff));
+  RoundedButtonStyle({
+    this.height,
+    this.width,
+    this.iconButtonColor,
+    this.backgroundColor,
+    this.iconEdgePadding,
+    this.buttonShape,
+    this.borderRadius,
+    this.edgePadding,
+    this.buttonTextStyle,
+    this.buttonIconSize,
+  });
+
+  factory RoundedButtonStyle.defaultStyle() {
+    return RoundedButtonStyle(
+      iconButtonColor: Color(0xFFFFFFFF),
+      backgroundColor: Color(0xff24d900),
+      buttonShape: BoxShape.rectangle,
+      borderRadius: BorderRadius.all(Radius.circular(20)),
+      edgePadding: EdgeInsets.all(0),
+      buttonTextStyle: null,
+      height: 24.0,
+      width: 24.0,
+      iconEdgePadding: 0,
+      buttonIconSize: 9.0,
+    );
+  }
 
-  final double iconEdgePadding = 5;
-  final double size = 56.0;
-  final double buttonIconSize = null;
+  factory RoundedButtonStyle.bigRoundedButton() {
+    return RoundedButtonStyle.defaultStyle().copyWith(
+        borderRadius: BorderRadius.all(Radius.circular(40)),
+        height: 48.0,
+        width: 48.0,
+        buttonIconSize: 14.0);
+  }
 
-  const _DefaultStyle();
-}
+  factory RoundedButtonStyle.largeRoundedButtonStyle() {
```

Comment:
Add a TODO comment to refactor this on another PR style refactor. This should be outside of this class should only have the default one